src/scripts/main.js

```diff
@@ -1,7 +1,212 @@
 'use strict';
+/* eslint-disable */
+const refs = {
+  startBtn: document.querySelector('.start'),
+  fieldCells: document.querySelectorAll('.field-cell'),
+  fieldRows: document.querySelectorAll('.field-row'),
+  messageStart: document.querySelector('.message-start'),
+  messageLose: document.querySelector('.message-lose'),
+  messageWin: document.querySelector('.message-win'),
+  score: document.querySelector('.game-score'),
+};
 
-// Uncomment the next lines to use your game instance in the browser
-// const Game = require('../modules/Game.class');
-// const game = new Game();
+const WIN_VALUE = 2048;
+const FIELD_LENGTH = 15;
+const NEW_CELL_VALUE = 2;
+const RARE_NEW_CELL_VALUE = 4;
+const RARE_CELL_CHANCE = 10;
 
-// Write your code here
+const inputHandler = ({ key }) => {
+  moveCells(key);
+};
+
+refs.startBtn.addEventListener('click', newGame);
+
+function newGame() {
+  resetField();
+  addCell(getRandomCell(), getToken());
+  addCell(getRandomCell(), getToken());
+  window.addEventListener('keydown', inputHandler);
+}
+
+function getRandomCell() {
+  if (!hasFreeSpace()) {
+    return -1;
+  }
+
+  let randomCell = getRandomNum(FIELD_LENGTH);
+
+  while (refs.fieldCells[randomCell].textContent !== '') {
+    randomCell = getRandomNum(FIELD_LENGTH);
+  }
+
+  return randomCell;
+}
+
+function getToken() {
+  return getRandomNum(100) <= RARE_CELL_CHANCE
+    ? RARE_NEW_CELL_VALUE
+    : NEW_CELL_VALUE;
+}
+
+function getRandomNum(maxVal) {
+  return Math.floor(Math.random() * maxVal);
+}
+
+function hasFreeSpace() {
+  return [...refs.fieldCells].some((cell) => cell.textContent === '');
+}
+
+function resetField() {
+  const { messageStart, messageLose, messageWin, score, fieldCells } = refs;
+
+  score.textContent = '0';
+
+  [messageStart, messageLose, messageWin].forEach(({ classList }) =>
+    classList.add('hidden'),
+);
+
+  fieldCells.forEach((cell) => {
```

Comment:
Directly accessing classList[1] assumes a specific class order which can be error-prone. It's safer to remove a specific class by name.